src/json_interface.c

```diff
@@ -25,169 +25,166 @@
-extern FILE *g_file_handle;
-
 typedef struct __ll_node {
     char *entry;
     char *value;
     struct __ll_node *next;
-} ji_ll_t;
+} ll_t;
 
 /*----------------------------------------------------------------------------*/
 /*                             File Scoped Variables                          */
 /*----------------------------------------------------------------------------*/
-static ji_ll_t *head = NULL;
-static bool cjson_init = false;
+static char *f_name = NULL;
+static ll_t *head = NULL;
 
 /*----------------------------------------------------------------------------*/
 /*                             Function Prototypes                            */
 /*----------------------------------------------------------------------------*/
-static jir_t __ji_persist(void);
-static char *__ji_cjson_create(ji_ll_t *node);
-static void  __ji_cjson_init(void);
+static int __add_node(const char *entry, const char *value);
 
 /*----------------------------------------------------------------------------*/
 /*                             External Functions                             */
 /*----------------------------------------------------------------------------*/
-jir_t ji_add_entry( const char *entry, const char *value )
+int ji_init(const char *file_name)
 {
-    ji_ll_t *current = head;
-
-    while( NULL != current )
-    {
-        if( 0 == strcmp( entry, current->entry ) )
-        {
-            if( 0 == strcmp( value, current->value ) )
-            {
-                return JIRT__ENTRY_ALREADY_PRESENT;
-            } 
-            else
-            {
-                free(current->value);
-                current->value = strdup(value); 
-                return __ji_persist();
-            }
-        }
-        current = current->next;
-    }
-
-    current = (ji_ll_t *) malloc( sizeof(ji_ll_t) );
-    current->entry = strdup(entry);
-    current->value = strdup(value);
-    current->next = NULL;
+    size_t length;
+    int read_size;
+    char *buf;
+    char *service, *url;
+    FILE *file_handle = fopen(file_name, "r");
+    cJSON_Hooks cjhooks;
 
-    if( NULL == head ) {
-        head = current;
+    if( NULL == file_handle ) {
+        return EXIT_FAILURE;
     }
 
-    __ji_persist();
-    return JIRT__SUCCESS;
-}
-
-jir_t ji_retrieve_entry( const char *entry, char **object )
-{
-    ji_ll_t *current = head;
-
-    while( NULL != current )
-    {
-        if( 0 == strcmp(entry, current->entry) )
-        {
-            *object = __ji_cjson_create(current);
-            return JIRT__SUCCESS;
-        }
-        current = current->next;
+    f_name = strdup(file_name);
+    read_size = getline(&buf, &length, file_handle);
+    while( -1 != read_size ) {
+        char *c = strchr(buf, COMMA);
+        service = strndup(buf, (c - buf));
+        url = strndup((c + 1), &buf[length-1] - c);
+        __add_node(service, url);
+        free(service); free(url);
```

Comment:
__add_node() is also being used in another place, want to keep ll behavior in one place. The code is the way it is here because of using strchr.